File: 000_MaxSumSubarrays/test00.cpp
/* Maimum value of all possible subarrays
Time complexity O(n^2)*/
#include <climits>
#include <iostream>

using namespace std;

int MaimumSumSubarray(int *arr, int n);

int main() {
  int arr[] = {3, -2, 5, -1};
  cout << MaimumSumSubarray(arr, sizeof(arr) / sizeof(arr[0])) << "\n";
  return 0;
}

int MaimumSumSubarray(int *arr, int n) {
  int ans = INT_MIN, startIndex, subArraySize;

  for (subArraySize = 1; subArraySize <= n; subArraySize++) {
    for (startIndex = 0; startIndex < n; startIndex++) {
      if ((startIndex + subArraySize) > n) {
        break;
      }
      int sum = 0;
      for (int i = 0; i < (startIndex + subArraySize); i++) {
        sum += arr[i];
      }
      ans = max(ans, sum);
    }
  }
  return ans;
}

File: 000_MaxSumSubarrays/test00.py
'''
 * @file test00.py
 * @author M. Z.
 * @brief Maximum value of all subarrays:Time complexity O(n^2)
 * @version 0.1
 * @date 2025-02-14
 *
 * @copyright Copyright (c) 2025
'''

def MaximumSumSubarray(arr, sizeArr):
    ret= -float('inf')
    for subArrSize in range(1, sizeArr + 1):
        for startIndex in range(0, sizeArr):
            if (startIndex + subArrSize) > sizeArr:
                break
            sum = 0
            for i in range (0, startIndex + subArrSize):
                sum += arr[i]
            ret = sum if (sum > ret) else ret
    return ret

arr =[3,-2,5,-1,4]

print("Size of array is", len(arr), "and maximum vaue of all subarrays is", MaximumSumSubarray(arr,len(arr)))

